src/tools/perf/lib/uct_tests.cc

```diff
@@ -398,16 +413,29 @@ class uct_perf_test_runner {
             } else{
                 send_sn     = 0; /* Remote buffer will remain 0 throughout the test */
             }
-            *(psn_t*)buffer = send_sn;
+
+            m_perf.allocator->memcpy(buffer, m_perf.uct.send_mem.mem_type,
+                                     &send_sn, UCS_MEMORY_TYPE_HOST,
+                                     sizeof(send_sn));
 
             UCX_PERF_TEST_FOREACH(&m_perf) {
                 if (flow_control) {
                     /* Wait until getting ACK from responder */
-                    ucs_assertv(UCS_CIRCULAR_COMPARE8(send_sn - 1, >=, *recv_sn),
-                                "recv_sn=%d iters=%ld", *recv_sn, m_perf.current.iters);
-                    while (UCS_CIRCULAR_COMPARE8(send_sn, >, (psn_t)(*recv_sn + fc_window))) {
+                    m_perf.allocator->memcpy(&sn, UCS_MEMORY_TYPE_HOST,
+                                             const_cast<const psn_t*>(recv_sn),
+                                             m_perf.uct.recv_mem.mem_type,
+                                             sizeof(*recv_sn));
+                    ucs_assertv(UCS_CIRCULAR_COMPARE8(send_sn - 1, >=, sn),
+                                "recv_sn=%d iters=%ld", sn, m_perf.current.iters);
+
+                    do {
                         progress_responder();
-                    }
+                        m_perf.allocator->memcpy(&sn, UCS_MEMORY_TYPE_HOST,
```

Comment:
there is some code duplication.. maybe ad helper function like `psn_t get_recv_sn()`?

Reply:
done